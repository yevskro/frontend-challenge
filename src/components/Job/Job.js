import React from 'react';
import styled from 'styled-components';

function Job(props){
  return <StyledJob featured={props.featured}></StyledJob>
}

const StyledJob = styled.li`
  box-sizing: border-box;
  display: flex;
  align-items: center;
  border-radius: 5px;
  border-left: 5px solid ${props => props.featured ? '#5BA6A2' : '#FFF'};
  background-color: #FFF;
  box-shadow: 0px 10px 10px RGBA(94, 140, 160, 0.15);
  font-family: Spartan;
  font-size: 13px;
  font-weight: 600;
  width: 100px;
  height: 100px;
  margin-top: 56px;
  width: 100%;
`;

export default Job;