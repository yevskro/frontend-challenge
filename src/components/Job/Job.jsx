import React from 'react';
import styled from 'styled-components';
import Company from '../Company/Company';
import Position from '../Position/Position';
import Filters from '../Filters/Filters';
import Filter from '../Filter/Filter';

function Job({ data }) {
  return (
    <StyledJob featured={data.featured}>
      <Logo src={data.logo} />
      <Information>
        <Company newly={data.newly} featured={data.featured}>
          {data.company}
        </Company>
        <Position
          postedAt={data.postedAt}
          contract={data.contract}
          location={data.location}
        >
          {data.position}
        </Position>
        <HorizontalLine />
        <Filters>
          <Filter>Frontend</Filter>
          <Filter>Junior</Filter>
          <Filter>CSS</Filter>
          <Filter>JavaScript</Filter>
        </Filters>
      </Information>
    </StyledJob>
  );
}

const HorizontalLine = styled.hr`
  border-top: 1px solid rgb(191, 193, 192, 0.75);
`;

const Information = styled.div`
  padding-left: 20px;
  padding-right: 24px;
  padding-top: 32px;
`;

const Logo = styled.img`
  width: 48px;
  height: 48px;
  position: absolute;
  transform: translate(
    19px,
    -24px
  ); /* take account of the border width of 5px from x */
`;

const StyledJob = styled.div`
  box-sizing: border-box;
  border-radius: 5px;
  border-left: 5px solid ${(props) => (props.featured ? '#5BA6A2' : '#FFF')};
  background-color: #fff;
  box-shadow: 0px 10px 10px RGBA(94, 140, 160, 0.15);
  font-family: Spartan;
  font-size: 13px;
  font-weight: 600;
  margin-top: 40px;
  width: 100%;
  padding-bottom: 14px;
`;

export default Job;
