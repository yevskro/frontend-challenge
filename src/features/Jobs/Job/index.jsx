import React from 'react';
import styled from 'styled-components';
import JobPosition from '../JobPosition';
import SearchFilters from '../../SearchByFilters/SearchFilters';
import SearchFilter from '../../SearchByFilters/SearchFilter';

function Job({ data, onAddFilter }) {
  function jsxFilters() {
    /* creates SearchFilters based on data information */
    const filters = [];

    filters.push(
      <SearchFilter
        onAddFilter={onAddFilter}
        filter={{ role: data.role }}
        key={`job-item-filter-${data.role}`}
      >
        {data.role}
      </SearchFilter>
    );

    filters.push(
      <SearchFilter
        onAddFilter={onAddFilter}
        filter={{ level: data.level }}
        key={`job-item-filter-${data.level}`}
      >
        {data.level}
      </SearchFilter>
    );

    data.languages.forEach((language) => {
      filters.push(
        <SearchFilter
          onAddFilter={onAddFilter}
          filter={{ languages: language }}
          key={`job-item-filter-${language}`}
        >
          {language}
        </SearchFilter>
      );
    });

    data.tools.forEach((tool) => {
      filters.push(
        <SearchFilter
          onAddFilter={onAddFilter}
          filter={{ tools: tool }}
          key={`job-item-filter-${tool}`}
        >
          {tool}
        </SearchFilter>
      );
    });

    return filters;
  }

  return (
    <StyledJob featured={data.featured} role="listitem">
      <JobInformation>
        <CompanyLogo src={data.logo} alt="company logo" />
        <JobPosition
          company={{
            name: data.company,
            newly: data.newly,
            featured: data.featured,
          }}
          postedAt={data.postedAt}
          contract={data.contract}
          location={data.location}
        >
          {data.position}
        </JobPosition>
      </JobInformation>
      <HorizontalLine />
      <JobFilters>
        <SearchFilters flexEnd>{jsxFilters()}</SearchFilters>
      </JobFilters>
    </StyledJob>
  );
}

const JobFilters = styled.div`
  display: flex;
  align-items: center;
  margin-left: auto;
`;

const HorizontalLine = styled.hr`
  opacity: 0.5;
  margin: 0px 0px 4px 0px;
  border-top: 1px solid ${({ theme }) => theme.color.neutral.darkGrayishCyan};
  @media (min-width: 785px) {
    /* desktop view */
    display: none;
  }
`;

const JobInformation = styled.div`
  display: flex;
  align-items: center;
  padding-bottom: 10px;
  @media (min-width: 785px) {
    /* desktop view */
    padding-bottom: 0px;
  }
`;

const CompanyLogo = styled.img`
  position: absolute;
  height: 48px;
  width: 48px;
  transform: translate(
    0px,
    -86px
  ); /* translate the icon to align half way above the container top border */
  @media (min-width: 555px) {
    /* tablet view */
    position: relative;
    height: 88px;
    width: 88px;
    margin-right: 24px;
    transform: translate(0, 0);
  }
`;

const StyledJob = styled.div`
  box-sizing: border-box;
  width: 100%;
  margin-top: 40px;
  padding: 44px 24px 14px 20px;
  background-color: white;
  border-radius: 5px;
  border-left: 5px solid
    ${(props) =>
      props.featured ? props.theme.color.primary.desaturtedDarkCyan : 'white'};
  box-shadow: 0px 10px 10px ${({ theme }) => theme.color.shadow};

  @media (min-width: 555px) {
    /* tablet view */
    display: flex;
    flex-direction: column;
    padding-top: 20px;
  }

  @media (min-width: 785px) {
    /* desktop view */
    display: flex;
    flex-direction: row;
    align-items: center;
    padding: 24px 24px 24px 24px;
  }

  @media (min-width: 900px) {
    /* fits it nicer on a bigger desktop view */
    padding: 30px 48px 30px 48px;
  }
`;

export default Job;
