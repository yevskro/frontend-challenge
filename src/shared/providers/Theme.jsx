import React from 'react';
import { ThemeProvider } from 'styled-components';
/* theme developed based on the style-guide */
const theme = {
  color: {
    primary: {
      desaturtedDarkCyan: 'hsl(180, 29%, 50%)',
    },
    neutral: {
      lightGrayishCyanBackground: 'hsl(180, 52%, 96%)',
      lightGrayishCyanFilterTablets: 'hsl(180, 31%, 95%)',
      darkGrayishCyan: 'hsl(180, 8%, 52%)',
      veryDarkGrayishCyan: 'hsl(180, 14%, 20%)',
    },
    shadow: 'rgba(94, 140, 160, 0.15)',
  },
  fontSize: {
    xxsmall: '0.7rem',
    xsmall: '0.8rem',
    small: '0.9rem',
    medium: '1rem',
    large: '1.2rem',
  },
  fontWeight: {
    thin: '400',
    normal: '500',
    bold: '600',
  },
  fontFamily: '"Spartan", "sans-serif"',
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
