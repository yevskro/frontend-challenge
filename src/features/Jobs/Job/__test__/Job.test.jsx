import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import jobData from '../../../../shared/test/test-data.json';
import Theme from '../../../../shared/providers/Theme';
import Job from '..';

/*
    "id": 1,
    "company": "Photosnap",
    "logo": "/assets/images/photosnap.svg",
    "newly": true,
    "featured": false,
    "position": "Senior Frontend Developer",
    "role": "Frontend",
    "level": "Senior",
    "postedAt": "2d ago",
    "contract": "Part Time",
    "location": "USA Only",
    "languages": ["HTML", "CSS", "JavaScript"],
    "tools": []
*/
test('renders', () => {
  render(
    <Theme>
      <Job data={jobData[0]} />
    </Theme>
  );
  expect(screen.getByText('Photosnap')).toBeInTheDocument();
  expect(screen.getByRole('img', { name: 'company logo' })).toHaveAttribute(
    'src',
    '/assets/images/photosnap.svg'
  );
  expect(screen.getByRole('img', { name: 'company logo' })).toBeInTheDocument();
  expect(screen.getByText('NEW!')).toBeInTheDocument();
  expect(screen.getByText('Senior Frontend Developer')).toBeInTheDocument();
  expect(screen.getByText('Frontend')).toBeInTheDocument();
  expect(screen.getByText('Senior')).toBeInTheDocument();
  expect(screen.getByText('2d ago')).toBeInTheDocument();
  expect(screen.getByText('Part Time')).toBeInTheDocument();
  expect(screen.getByText('USA Only')).toBeInTheDocument();
  expect(screen.getByText('HTML')).toBeInTheDocument();
  expect(screen.getByText('CSS')).toBeInTheDocument();
  expect(screen.getByText('JavaScript')).toBeInTheDocument();
});

test('snapshot', async () => {
  const tree = renderer
    .create(
      <Theme>
        <Job data={jobData[0]} />
      </Theme>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
