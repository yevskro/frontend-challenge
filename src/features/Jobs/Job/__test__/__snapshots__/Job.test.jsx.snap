// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c10 {
  white-space: nowrap;
}

.c10:first-child::before {
  font-size: 1.7rem;
  content: ' ';
}

.c10:not(:first-child)::before {
  display: inline-block;
  content: '.';
  padding: 0px 4px 0px 4px;
  font-size: 1.7rem;
  -webkit-transform: translateY(-16%);
  -ms-transform: translateY(-16%);
  transform: translateY(-16%);
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: hsl(180,8%,52%);
  font-weight: 400;
  font-size: 0.9rem;
}

.c5 {
  height: 24px;
  line-height: 27px;
  vertical-align: middle;
  padding: 0px 9px 0px 9px;
  margin: 0px 4px 0px 4px;
  color: white;
  border-radius: 20px;
  font-size: 0.9rem;
}

.c6 {
  margin-left: 18px;
  background-color: hsl(180,29%,50%);
  font-size: 0.7rem;
}

.c7 {
  background-color: hsl(180,14%,20%);
  font-size: 0.7rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 24px;
  color: hsl(180,29%,50%);
  font-weight: 600;
  font-size: 0.8rem;
}

.c8 {
  padding: 11px 0px 11px 0px;
  margin: 0px;
  white-space: nowrap;
  font-weight: 600;
}

.c8:hover {
  color: hsl(180,29%,50%);
  cursor: pointer;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  margin: 10px 16px 10px 0px;
  border-radius: 5px;
  overflow: hidden;
  background-color: hsl(180,31%,95%);
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  padding: 0px 9px 0px 8px;
  color: hsl(180,29%,50%);
  font-size: 0.9rem;
  font-weight: 600;
}

.c15:hover {
  color: white;
  background-color: hsl(180,29%,50%);
  cursor: pointer;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: auto;
}

.c11 {
  opacity: 0.5;
  margin: 0px 0px 4px 0px;
  border-top: 1px solid hsl(180,8%,52%);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 10px;
}

.c2 {
  position: absolute;
  height: 48px;
  width: 48px;
  -webkit-transform: translate( 0px, -86px );
  -ms-transform: translate( 0px, -86px );
  transform: translate( 0px, -86px );
}

.c0 {
  box-sizing: border-box;
  width: 100%;
  margin-top: 40px;
  padding: 44px 24px 14px 20px;
  background-color: white;
  border-radius: 5px;
  border-left: 5px solid hsl(180,29%,50%);
  box-shadow: 0px 10px 10px rgba(94,140,160,0.15);
}

@media (min-width:600px) {
  .c10:not(:first-child)::before {
    padding: 0px 8px 0px 8px;
  }
}

@media (min-width:600px) {
  .c10 {
    font-size: 1rem;
  }
}

@media (min-width:500px) {
  .c4 {
    font-size: 0.9rem;
  }
}

@media (min-width:500px) {
  .c3 {
    font-size: 1rem;
  }
}

@media (min-width:600px) {
  .c3 {
    font-size: 1.2rem;
  }
}

@media (max-width:415px) {
  .c13 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media (min-width:415px) {
  .c14 {
    margin: 10px 0px 10px 16px;
  }
}

@media (min-width:785px) {
  .c11 {
    display: none;
  }
}

@media (min-width:785px) {
  .c1 {
    padding-bottom: 0px;
  }
}

@media (min-width:555px) {
  .c2 {
    position: relative;
    height: 88px;
    width: 88px;
    margin-right: 24px;
    -webkit-transform: translate(0,0);
    -ms-transform: translate(0,0);
    transform: translate(0,0);
  }
}

@media (min-width:555px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    padding-top: 20px;
  }
}

@media (min-width:785px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    padding: 24px 24px 24px 24px;
  }
}

@media (min-width:900px) {
  .c0 {
    padding: 30px 48px 30px 48px;
  }
}

<div
  className="c0"
  role="listitem"
>
  <div
    className="c1"
  >
    <img
      alt="company logo"
      className="c2"
      src="/assets/images/photosnap.svg"
    />
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <h6>
          Photosnap
        </h6>
        <span
          className="c5 c6"
        >
          NEW!
        </span>
        <span
          className="c5 c7"
        >
          FEATURED
        </span>
      </div>
      <h3
        className="c8"
      >
        Senior Frontend Developer
      </h3>
       
      <ul
        className="c9"
      >
        <li
          className="c10"
        >
          2d ago
        </li>
        <li
          className="c10"
        >
          Part Time
        </li>
        <li
          className="c10"
        >
          USA Only
        </li>
      </ul>
    </div>
  </div>
  <hr
    className="c11"
  />
  <div
    className="c12"
  >
    <ul
      className="c13"
    >
      <li
        className="c14"
      >
        <span
          className="c15"
          onClick={[Function]}
        >
          Frontend
        </span>
        
      </li>
      <li
        className="c14"
      >
        <span
          className="c15"
          onClick={[Function]}
        >
          Senior
        </span>
        
      </li>
      <li
        className="c14"
      >
        <span
          className="c15"
          onClick={[Function]}
        >
          HTML
        </span>
        
      </li>
      <li
        className="c14"
      >
        <span
          className="c15"
          onClick={[Function]}
        >
          CSS
        </span>
        
      </li>
      <li
        className="c14"
      >
        <span
          className="c15"
          onClick={[Function]}
        >
          JavaScript
        </span>
        
      </li>
    </ul>
  </div>
</div>
`;
